### ISSUE TITLE
Default values in sub models are overridden by env vars of other values

### ISSUE THREAD

**Issue by abompard** [pydantic/pydantic#5266]
### Initial Checks

- [X] I have searched GitHub for a duplicate issue and I'm sure this is something new
- [X] I have searched Google & StackOverflow for a solution and couldn't find anything
- [X] I have read and followed [the docs](https://docs.pydantic.dev) and still think this is a bug
- [X] I am confident that the issue is with pydantic (not my code, or another library in the ecosystem like [FastAPI](https://fastapi.tiangolo.com) or [mypy](https://mypy.readthedocs.io/en/stable))

### Description

This is for settings management. When a submodel is set with default values, setting one of those values by an environment variable overrides the other values as well.

### Example Code

```Python
import os
from pydantic import BaseModel, BaseSettings


class SubModel(BaseModel):
    v1: str = "v1"
    v2: str

class Settings(BaseSettings):
    sub_model: SubModel = SubModel(v2="v2")

    class Config:
        env_nested_delimiter = '__'

# This works
print(Settings().dict())
# Now set submodel.v1
os.environ.update(dict(
    SUB_MODEL__V1="v1.2",
))
# This fails because submodel.v2 is not defined anymore
print(Settings().dict())
```

### Python, Pydantic & OS Version

```Text
pydantic version: 1.10.7
            pydantic compiled: True
                 install path: [redacted]/lib64/python3.10/site-packages/pydantic
               python version: 3.10.5 (main, Jun  9 2022, 00:00:00) [GCC 12.1.1 20220507 (Red Hat 12.1.1-1)]
                     platform: Linux-5.17.5-300.fc36.x86_64-x86_64-with-glibc2.35
     optional deps. installed: ['dotenv', 'typing-extensions']
```

### Affected Components

- [ ] [Compatibility between releases](https://docs.pydantic.dev/changelog/)
- [X] [Data validation/parsing](https://docs.pydantic.dev/usage/models/#basic-model-usage)
- [ ] [Data serialization](https://docs.pydantic.dev/usage/exporting_models/) - `.dict()` and `.json()`
- [ ] [JSON Schema](https://docs.pydantic.dev/usage/schema/)
- [ ] [Dataclasses](https://docs.pydantic.dev/usage/dataclasses/)
- [ ] [Model Config](https://docs.pydantic.dev/usage/model_config/)
- [ ] [Field Types](https://docs.pydantic.dev/usage/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://docs.pydantic.dev/usage/validation_decorator/)
- [ ] [Generic Models](https://docs.pydantic.dev/usage/models/#generic-models)
- [ ] [Other Model behaviour](https://docs.pydantic.dev/usage/models/) - `construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://docs.pydantic.dev/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by hramezani** [comment]
I think this is the right behavior because by setting the env variables you are replacing the default. Actually it's like you are initializing model like:
```py
Settings(**{'sub_model': {'v1': 'v1.2'}})
```
which means creating `SubModel` object without `v2` (`SubModel(v1='v1.2')`)

**Comment by abompard** [comment]
Yeah, I get why it's doing this, but it seems unintuitive that a default would be replaced when I didn't explicitly change it.

What should I do then, if I want to change only one value in a submodel? Set the environment variables to the default values for all the other keys?

**Comment by hramezani** [comment]
You can define the default values in model as well
